pkg/resource/google/google_bigquery_dataset_test.go

```diff
@@ -18,10 +18,15 @@ func TestAcc_Google_BigqueryDataset(t *testing.T) {
 		},
 		Checks: []acceptance.AccCheck{
 			{
-				// New resources are not visible immediately on GCP api after an apply operation
-				// Logic below retry driftctl scan until we can retrieve the results (infra will be in sync) and for maximum 60 seconds
+				// New resources are not visible immediately through GCP API after an apply operation.
+				// Logic below retries driftctl scan using a back-off strategy of retrying 'n' times
+				// and doubling the amount of time waited after each one.
 				ShouldRetry: func(result *test.ScanResult, retryDuration time.Duration, retryCount uint8) bool {
-					return !result.IsSync() && retryDuration < time.Minute
+					if result.IsSync() || retryDuration > 10*time.Minute {
```

Comment:
```suggestion
					if result.IsSync() || retryDuration < 10*time.Minute {
```

I think the direction of the inequality was correct before, right? We want to abort the retry loop if we've been retrying for longer than retryDuration, if I understand the code that calls this function correctly: https://github.com/snyk/driftctl/blob/b751cd50aefb2af820d58c589211ff0371780107/test/acceptance/testing.go#L436-L440